test/functional/p2p_invalid_block.py

```diff
@@ -133,5 +136,19 @@ def run_test(self):
         self.log.info("Test inflation by duplicating input")
         peer.send_blocks_and_test([block4], node, success=False,  reject_reason='bad-txns-inputs-duplicate')
 
+
+        self.log.info("Test accepting identical block after rejecting it due to a future timestamp.")
+
+        t = int(time.time())
+        node.setmocktime(t)
+        block = create_block(tip, create_coinbase(height), t + 7201)
```

Comment:
This is 1 second away from the cutoff right? Perhaps a slightly larger buffer would be better because the functional tests usually run multithreaded.

Reply:
Another small suggestion, using `2 * 60 * 60` or `MAX_FUTURE_BLOCK_TIME` instead of a magic number might be a bit more clear

Reply:
I went for `MAX_FUTURE_BLOCK_TIME` in the end and added a new `t_premature = 60`, so that we have a 60 second buffer to avoid any potential issues.